scripts/events_module/event_filters.py

```diff
@@ -0,0 +1,412 @@
+        if "low" in trigger and herb_supply[chosen_herb] < half_amount:
+            return True
+        elif "adequate" in trigger and half_amount < herb_supply[chosen_herb] <= needed_amount:
+            return True
+        elif "full" in trigger and needed_amount < herb_supply[chosen_herb] <= needed_amount * 2:
+            return True
+        elif "excess" in trigger and needed_amount * 2 < herb_supply[chosen_herb]:
+            return True
+
+        return False
+
+
+def event_for_cat(cat_info: dict, cat, cat_group: list = None, event_id: str = None, p_l=None) -> bool:
+    """
+        checks if a cat is suitable for the event
+        :param cat_info: cat's dict of constraints
+        :param cat: the cat object of the cat being checked
+        :param cat_group: the group of cats being included within the event
+        :param event_id: if event comes with an id, include it here
+        :param p_l: if event is a patrol, include patrol leader object here
+        """
+
+    func_lookup = {
+        "age": _check_cat_age(cat, cat_info.get("age", [])),
+        "status": _check_cat_status(cat, cat_info.get("status", [])),
+        "trait": _check_cat_trait(cat, cat_info.get("trait", []), cat_info.get("not_trait", [])),
+        "skills": _check_cat_skills(cat, cat_info.get("skill", []), cat_info.get("not_skill", [])),
+        "backstory": _check_cat_backstory(cat, cat_info.get("backstory", [])),
+        "gender": _check_cat_gender(cat, cat_info.get("gender", []))
+    }
+
+    for func in func_lookup:
+        if not func_lookup[func]:
+            return False
+
+    if cat_info.get("relationship_status", []):
+        if not filter_relationship_type(
+                group=cat_group,
+                filter_types=cat_info["relationship_status"],
+                event_id=event_id,
+                patrol_leader=p_l
+        ):
+            return False
+
+    return True
+
+
+def _check_cat_age(cat, ages: list) -> bool:
+    """
+        checks if a cat's age is within ages list
+        """
+    if "any" in ages or not ages:
+        return True
+
+    if cat.age in ages:
+        return True
+
+    return False
+
+
+def _check_cat_status(cat, statuses: list) -> bool:
+    """
+        checks if cat's status is within statuses list
+        """
+    if "any" in statuses or not statuses:
+        return True
+
+    if cat.status in statuses:
+        return True
+
+    return False
+
+
+def _check_cat_trait(cat, traits: list, not_traits: list) -> bool:
+    """
+    checks if cat has the correct traits for traits and not_traits lists
+    """
+    if not traits and not not_traits:
+        return True
+
+    cat_trait = cat.personality.trait
+    allowed = False
+
+    if traits:
+        if cat_trait in traits:
+            allowed = True
+        else:
+            return False
+    if not_traits:
+        if cat_trait in not_traits:
+            allowed = False
+        else:
+            return True
+
+    return allowed
```

Comment:
```suggestion
    if traits and cat_trait not in traits:
        return False
    if not_traits and cat_trait in not_traits:
        return False
    return True
```
I will kill all your interim variables if it kills me /silly